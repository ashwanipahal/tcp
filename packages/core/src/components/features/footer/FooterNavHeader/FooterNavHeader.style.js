import { css } from 'styled-components';

const GlobalCss = css`
  font-family: Nunito-SemiBold;
  font-size: 16px;
  color: #1a1a1a;
  ::after {
    content: '';
    height: 1px;
    width: 80px;
    background: #439ad4;
    border: 10px;
    display: block;
    margin-top: 26px;
  }
`;

export default GlobalCss;
