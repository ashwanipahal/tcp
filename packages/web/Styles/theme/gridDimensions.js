export default {
  gridBreakPointsKeys: ['small', 'medium', 'large', 'xlarge'],
  numberOfColumnsObj: {
    small: 6,
    medium: 8,
    large: 12,
    xlarge: 12,
  },
  columnWidthObj: {
    small: 42,
    medium: 66,
    large: 77,
    xlarge: 90,
  },
  columnGutterObj: {
    small: 19,
    medium: 30,
    large: 22,
    xlarge: 30,
  },
  gridOffsetObj: {
    small: 14,
    medium: 15,
    large: 14,
    xlarge: 15,
  },
  availableMaxWidthObj: {
    small: 347,
    medium: 740,
    large: 996,
    xlarge: 1410,
  },
};
