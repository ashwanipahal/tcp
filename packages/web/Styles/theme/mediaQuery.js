export const BREAKPOINTS = {
  small: '375px',
  smallMax: '767px',
  medium: '768px',
  mediumMax: '1023px',
  large: '1024px',
  largeMax: '1439px',
  xlarge: '1440px',
};

export const MEDIA_QUERIES = {
  small: `(min-width: ${BREAKPOINTS.small})`,
  smallMax: `(max-width: ${BREAKPOINTS.smallMax})`,
  smallOnly: `
    (min-width: ${BREAKPOINTS.small})
    and
    (max-width: ${BREAKPOINTS.smallMax})
  `,
  medium: `(min-width: ${BREAKPOINTS.medium})`,
  mediumMax: `(max-width: ${BREAKPOINTS.mediumMax})`,
  mediumOnly: `
    (min-width: ${BREAKPOINTS.medium})
    and
    (max-width: ${BREAKPOINTS.mediumMax})
  `,
  large: `(min-width: ${BREAKPOINTS.large})`,
  largeMax: `(max-width: ${BREAKPOINTS.largeMax})`,
  largeOnly: `
    (min-width: ${BREAKPOINTS.large})
    and
    (max-width: ${BREAKPOINTS.largeMax})
  `,
  xlarge: `(min-width: ${BREAKPOINTS.xlarge})`,
};
