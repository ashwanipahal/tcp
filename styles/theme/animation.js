const FADE_IN = `
0% {
  opacity: 0;
}

10% {
  opacity: 0;
}

100% {
  opacity: 1;
}
`;

const FADE_IN_UP = `
0% {
  opacity: 0;
  transform: translate3d(0, 20px, 0);
}

10% {
  opacity: 0;
  transform: translate3d(0, 20px, 0);
}

100% {
  opacity: 1;
  transform: translate3d(0, 0, 0);
}
`;

const FADE_IN_UP_LONG = `
0% {
  opacity: 0;
  transform: translate3d(0, 80px, 0);
}

10% {
  opacity: 0;
  transform: translate3d(0, 80px, 0);
}

100% {
  opacity: 1;
  transform: translate3d(0, 0, 0);
}
`;

const EXPAND = `
0% {
  opacity: 0;
  letter-spacing: 16px;
}

70% {
  opacity: 1;
}

100% {
  opacity: 1;
  letter-spacing: 18px;
}
`;

const EXPAND_SMALL = `
0% {
  opacity: 0;
  letter-spacing: 4px;
}

70% {
  opacity: 1;
}

100% {
  opacity: 1;
  letter-spacing: 8px;
}
`;

const SCALE_DOWNLOAD = {
  arrow: `
    0%{
      transform-origin: bottom;
      transform: scale(1);
    }

    50% {
      transform-origin: bottom;
      transform: scale(0);
    }

    85% {
      transform-origin: top;
      transform: scale(0);
    }

    100% {
      transform-origin: top;
      transform: scale(1);
    }
  `,
  stroke: `
    0%{
      transform: scaleY(1.5);
      transform-origin: bottom;
    }
    50% {
      transform: scaleY(0);
      top: auto;
      bottom: 0px;
      transform-origin: bottom;
    }
    50.1% {
      transform: scaleY(0);
      top: calc(100% - 24px);
      bottom: auto;
      transform-origin: top;
    }
    100% {
      transform: scaleY(1.5);
      top: calc(100% - 24px);
      transform-origin: top;
    }
  `,
};

const SCALE_DOWN_RIGHT = {
  arrow: `
    0% {
      transform-origin: right;
      transform: scale(1);
    }

    50% {
      transform-origin: right;
      transform: scale(0);
    }

    85% {
      transform-origin: left;
      transform: scale(0);
    }

    100% {
      transform-origin: left;
      transform: scale(1);
    }
  `,
  stroke: {
    small: `
      0% {
        transform: scaleX(1.3);
        transform-origin: right;
      }
      50% {
        transform: scaleX(0);
        right: 0px;
        left: auto;
        transform-origin: right;
      }
      50.1% {
        transform: scaleX(0);
        right: auto;
        left: calc(100% - 28px);
        transform-origin: left;
      }
      100% {
        transform: scaleX(1.3);
        left: calc(100% - 28px);
        transform-origin: left;
      }
    `,
    large: `
      0% {
        transform: scaleX(1.5);
        transform-origin: right;
      }
      50% {
        transform: scaleX(0);
        right: 0px;
        left: auto;
        transform-origin: right;
      }
      50.1% {
        transform: scaleX(0);
        right: auto;
        left: calc(100% - 32px);
        transform-origin: left;
      }
      100% {
        transform: scaleX(1.5);
        left: calc(100% - 32px);
        transform-origin: left;
      }
    `,
  },
};

const CLIP_REVEAL_LTR = `
0% {
  clip-path: polygon(0 0, 0 0, 0 100%, 0 100%);
}

100% {
  clip-path: polygon(0 0, 100% 0, 100% 100%, 0 100%);
}
`;

const CLIP_REVEAL_RTL = `
0% {
  clip-path: polygon(100% 0, 100% 0, 100% 100%, 100% 100%);
}

100% {
  clip-path: polygon(0 0, 100% 0, 100% 100%, 0 100%);
}
`;

const HINT_CLIP_REVEAL = `
0% {
  clip-path: polygon(95% 0, 100% 0, 100% 100%, 95% 100%);
}

100% {
  clip-path: polygon(0 0, 100% 0, 100% 100%, 0 100%);
}
`;

const HINT_CLIP_HIDE = `
0% {
  clip-path: polygon(95% 0, 100% 0, 100% 100%, 95% 100%);
}

100% {
  clip-path: polygon(0 0, 100% 0, 100% 100%, 0 100%);
}
`;

const FULL_SWIPE_LTR = `
0% {
  opacity: 1;
  left: -2px;
  transform: translateX(-100%);
}

99.9% {
  left: -2px;
}

100% {
  opacity: 1;
  left: 0;
  transform: translateX(0);
}
`;

const FULL_SWIPE_RTL = `
0% {
  opacity: 1;
  left: 2px;
  transform: translateX(0);
}

99.9% {
  opacity: 1;
  left: 2px;
}

100% {
  opacity: 1;
  left: 0;
  transform: translateX(-100%);
}
`;

const IMG_PARALLAX_LTR = `
0% {
  transform: translate3d(-8%, 0, 0);
}

100% {
  transform: translate3d(-1%, 0, 0);
}
`;

const IMG_PARALLAX_RTL = `
0% {
  transform: translate3d(8%, 0, 0);
}

100% {
  transform: translate3d(-1%, 0, 0);
}
`;

const FILTER_CONTENT = `
0% {
  opacity: 1;
}

100% {
  opacity: 0;
}
`;

const TEXT_HIGHLIGHT = `
0% {
  background-position: 0 0;
}

100% {
  background-position: -100% 0;
}
`;

// Easing
export default {
  fadeIn: FADE_IN,
  fadeInUp: FADE_IN_UP,
  fadeInUpLong: FADE_IN_UP_LONG,
  expand: EXPAND,
  expandSmall: EXPAND_SMALL,
  scaleDownRight: SCALE_DOWN_RIGHT,
  scaleDownload: SCALE_DOWNLOAD,
  clipReveal: {
    ltr: CLIP_REVEAL_LTR,
    rtl: CLIP_REVEAL_RTL,
  },
  hintClip: {
    reveal: HINT_CLIP_REVEAL,
    hide: HINT_CLIP_HIDE,
  },
  fullSwipe: {
    ltr: FULL_SWIPE_LTR,
    rtl: FULL_SWIPE_RTL,
  },
  imgParallax: {
    ltr: IMG_PARALLAX_LTR,
    rtl: IMG_PARALLAX_RTL,
  },
  filterContent: FILTER_CONTENT,
  textHighlight: TEXT_HIGHLIGHT,
};
