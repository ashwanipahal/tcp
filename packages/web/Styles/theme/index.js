import colors from './colors';
import fonts from './fonts';
import zindex from './zindex';
import gridDimensions from './gridDimensions';
import { MEDIA_QUERIES, BREAKPOINTS } from './mediaQuery';

const theme = {
  colors,
  fonts,
  MEDIA_QUERIES,
  BREAKPOINTS,
  zindex,
  gridDimensions,
};

export default theme;
