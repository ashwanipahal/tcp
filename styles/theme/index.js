import fonts from './fonts';
import colors from './colors';
import zIndex from './zIndex';
import mediaQuery from './mediaQuery';
import animation from './animation';
import easing from './easing';
import duration from './duration';
import delay from './delay';
import clipping from './clipping';
import sizes from './sizes';
import padding from './padding';
import borders from './borders';
import scale from './scale';
import iconAnimations from './iconAnimations';

export default {
  colors,
  fonts,
  zIndex,
  mediaQuery,
  animation,
  easing,
  duration,
  delay,
  clipping,
  sizes,
  padding,
  borders,
  scale,
  iconAnimations,
};
