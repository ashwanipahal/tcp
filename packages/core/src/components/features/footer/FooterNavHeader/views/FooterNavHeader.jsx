// @flow
import React from 'react';
import BodyCopy from '../../../../common/atoms/BodyCopy';
import { getIconPath } from '../../../../../utils';
import Anchor from '../../../../common/atoms/Anchor';
import Image from '../../../../common/atoms/Image';
import styles from '../FooterNavHeader.style';
import withStyles from '../../../../common/hoc/withStyles';

type Props = {
  className: string,
  ariaLabel: string,
  headerAsImage: boolean,
  colNum: number,
  titleText: string,
  titleObj: Object,
  isSubHeader: boolean,
};

const FooterNavHeader = ({
  className,
  titleText,
  titleObj,
  ariaLabel,
  headerAsImage,
  isSubHeader,
  colNum,
}: Props) => {
  if (!headerAsImage) {
    return (
      <BodyCopy
        className={!isSubHeader ? className : `${className} subHeader`}
        aria-label={ariaLabel}
        data-index={colNum}
        data-locator={`col_heading_${colNum}`}
        component="p"
        fontFamily="secondary"
        fontWeight="semibold"
        fontSize="fs16"
        color="text.primary"
      >
        {titleText}
      </BodyCopy>
    );
  }
  return (
    <Anchor
      to={titleObj.url}
      className={`${className} img-link`}
      dataLocator={`col_heading_${colNum}`}
    >
      <Image alt={titleObj.image_alt} src={getIconPath(titleObj.class)} />
    </Anchor>
  );
};

export default withStyles(FooterNavHeader, styles);

export { FooterNavHeader as FooterNavHeaderVanilla };
