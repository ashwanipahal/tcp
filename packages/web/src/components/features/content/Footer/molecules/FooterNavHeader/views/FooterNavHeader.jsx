import React from 'react';
import PropTypes from 'prop-types';
import { Anchor, BodyCopy, Image } from '@tcp/core/src/components/common/atoms';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import { getIconPath } from '@tcp/core/src/utils';
import styles from '../FooterNavHeader.style';

const FooterNavHeader = ({
  className,
  titleText,
  titleObj,
  ariaLabel,
  headerAsImage,
  isSubHeader,
  colNum,
}) => {
  if (!headerAsImage) {
    return (
      <BodyCopy
        className={!isSubHeader ? className : `${className} subHeader`}
        aria-label={ariaLabel}
        data-index={colNum}
        data-locator={`col_heading_${colNum}`}
        component="p"
        fontFamily="secondary"
        fontWeight="semibold"
        fontSize="fs16"
        color="text.primary"
      >
        {titleText}
      </BodyCopy>
    );
  }
  return (
    <Anchor
      to={titleObj.url}
      aria-label={titleText}
      className={`${className} img-link`}
      dataLocator={`col_heading_${colNum}`}
    >
      <Image
        tabIndex="-1"
        aria-hidden="true"
        alt={titleObj.image_alt}
        src={getIconPath(titleObj.class)}
      />
    </Anchor>
  );
};

FooterNavHeader.propTypes = {
  className: PropTypes.string.isRequired,
  ariaLabel: PropTypes.string.isRequired,
  headerAsImage: PropTypes.bool.isRequired,
  colNum: PropTypes.number.isRequired,
  titleText: PropTypes.string.isRequired,
  titleObj: PropTypes.shape({}).isRequired,
  isSubHeader: PropTypes.bool.isRequired,
};

export default withStyles(FooterNavHeader, styles);

export { FooterNavHeader as FooterNavHeaderVanilla };
