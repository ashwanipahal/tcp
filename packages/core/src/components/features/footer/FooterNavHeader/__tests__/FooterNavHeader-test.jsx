import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import theme from '@tcp/core/styles/themes/TCP';
import { ThemeProvider } from 'styled-components';
import FooterNavHeader from '../views/FooterNavHeader';

describe('FooterNavHeader component', () => {
  it('renders default footer correctly', () => {
    const props = {
      className: 'footer-nav-header1',
      titleText: 'Title',
      ariaLabel: 'ABCD',
      isSubHeader: false,
      index: 2,
    };
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <FooterNavHeader {...props} />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });

  it('renders subheader variation correctly', () => {
    const props = {
      className: 'footer-nav-header2',
      titleText: 'Title',
      ariaLabel: 'ABCD',
      isSubHeader: true,
      index: 4,
    };
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <FooterNavHeader {...props} />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });

  it('renders image in header variation correctly', () => {
    const props = {
      className: 'footer-nav-header',
      titleText: 'Title',
      ariaLabel: 'ABCD',
      isSubHeader: false,
      headerAsImage: true,
      index: 3,
      titleObj: {
        text: '',
        url: '',
      },
    };
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <FooterNavHeader {...props} />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
