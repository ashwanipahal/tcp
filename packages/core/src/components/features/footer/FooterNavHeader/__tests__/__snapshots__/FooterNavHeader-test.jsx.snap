// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterNavHeader component renders default footer correctly 1`] = `
.c0 {
  font-family: Nunito-SemiBold,Arial,Helvetica,sans-serif;
  font-size: 16px;
  color: #1a1a1a;
  margin: 0 0 37px 0;
}

.c0::after {
  content: '';
  height: 1px;
  width: 80px;
  background: #439ad4;
  border: 10px;
  display: block;
  margin-top: 26px;
}

.c0.img-link::after {
  display: none;
}

.c0.img-link img {
  padding-bottom: 35px;
}

.c0.subHeader {
  margin-top: 24px;
}

<h4
  aria-label="ABCD"
  className="footer-nav-header1 c0"
  data-index={2}
>
  Title
</h4>
`;

exports[`FooterNavHeader component renders image in header variation correctly 1`] = `
.c1 {
  font-family: Nunito,Arial,Helvetica,sans-serif;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  font-family: Nunito-SemiBold,Arial,Helvetica,sans-serif;
  font-size: 16px;
  color: #1a1a1a;
  margin: 0 0 37px 0;
}

.c0::after {
  content: '';
  height: 1px;
  width: 80px;
  background: #439ad4;
  border: 10px;
  display: block;
  margin-top: 26px;
}

.c0.img-link::after {
  display: none;
}

.c0.img-link img {
  padding-bottom: 35px;
}

.c0.subHeader {
  margin-top: 24px;
}

<a
  className="footer-nav-header c0 img-link c1"
  data-locator=""
  href=""
  onClick={[Function]}
  target=""
  title=""
>
  <img
    className=""
    onError={null}
    sizes={null}
    src="/static/images/undefined"
    srcSet={null}
  />
</a>
`;

exports[`FooterNavHeader component renders subheader variation correctly 1`] = `
.c0 {
  font-family: Nunito-SemiBold,Arial,Helvetica,sans-serif;
  font-size: 16px;
  color: #1a1a1a;
  margin: 0 0 37px 0;
}

.c0::after {
  content: '';
  height: 1px;
  width: 80px;
  background: #439ad4;
  border: 10px;
  display: block;
  margin-top: 26px;
}

.c0.img-link::after {
  display: none;
}

.c0.img-link img {
  padding-bottom: 35px;
}

.c0.subHeader {
  margin-top: 24px;
}

<h4
  aria-label="ABCD"
  className="footer-nav-header2 c0 subHeader"
  data-index={4}
>
  Title
</h4>
`;
